### ISSUE TITLE
[Bug]: Combobox list incorrect focus state placement and color contrast

### ISSUE THREAD

**Issue by satyaminnekanti** [carbon-design-system/carbon#12941]
### Package

carbon-components

### Browser

Chrome

### Package version

v10

### Description

**Accessibility Issues with Combo-box:**

Go to [combo-box sample](https://react.carbondesignsystem.com/?path=/story/components-combobox--default)
select an Option (Option 2)
with an Option selected, open the combo-box options and navigate through them

1. Issue 1: Focus not visible/shown clearly when navigated to currently selected item in combo ( i meant there is no colour change to indicate that the currently selected item received focus). This is a failure of [WCAG Checkpoint 2.4.7. Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
2. Issue 2: Also in combo-box once an option is selected and user navigated back to combo-box again and press down arrow, focus should be on the currently selected option and when user press Down Arrow key again focus should go to the next option after the selected but instead it always start from the top

Please see attached Gif:
![ezgif com-gif-maker](https://user-images.githubusercontent.com/122359404/211532366-e70fa1aa-8406-4389-8f45-4af05b3b0a88.gif)

Please refer [ARIA Authoring Practices Guide for Combo-box](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) for reference

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-combobox--default

### Steps to reproduce

**Steps to reproduce:**

1. Go to [combo-box sample](https://react.carbondesignsystem.com/?path=/story/components-combobox--default)
2. select an Option (Option 2)
3. with an Option selected, open the combo-box options and navigate through them

**Issues:**

1. Issue 1: Focus not visible/shown clearly when navigated to currently selected item in combo ( i meant there is no colour change to indicate that the currently selected item received focus). This is a failure of [WCAG Checkpoint 2.4.7. Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
2. Issue 2: Also in combo-box once an option is selected and user navigated back to combo-box again and press down arrow, focus should be on the currently selected option and when user press Down Arrow key again focus should go to the next option after the selected but instead it always start from the top

Please see attached Gif:
![ezgif com-gif-maker](https://user-images.githubusercontent.com/122359404/211532904-1c3912eb-72a5-4eb8-9ebf-fb60addba99d.gif)

Please refer [ARIA Authoring Practices Guide for Combo-box](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) for reference

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for reporting this! #1 is a duplicate of https://github.com/carbon-design-system/carbon/issues/12720 and https://github.com/carbon-design-system/carbon/issues/12723

@mbgower For #2, should the focus always be placed on the currently selected item as being suggested here?

**Comment by tay1orjones** [comment]
> should the focus always be placed on the currently selected item as being suggested here? Does always placing the focus on the first item in the list fail any WCAG success criterion

Based on internal conversations, yes this should be the case. The actionable piece of this issue is to address the focus behavior when reopening the ComboBox after having selected an option. Focus should be placed on the selected option, not the first option.
